vunit/sim_if/xcelium.py

```diff
@@ -0,0 +1,427 @@
+        else:
+            self._cdslib = str(Path(cdslib).resolve())
+        self._hdlvar = hdlvar
+        self._cds_root_xrun = self.find_cds_root_xrun()
+        self._create_cdslib()
+
+    def find_cds_root_xrun(self):
+        """
+        Finds xrun cds root
+        """
+        return subprocess.check_output(
+            [str(Path(self._prefix) / "cds_root"), "xrun"]
+        ).splitlines()[0]
+
+    def find_cds_root_virtuoso(self):
+        """
+        Finds virtuoso cds root
+        """
+        try:
+            return subprocess.check_output(
+                [str(Path(self._prefix) / "cds_root"), "virtuoso"]
+            ).splitlines()[0]
+        except subprocess.CalledProcessError:
+            return None
+
+    def _create_cdslib(self):
+        """
+        Create the cds.lib file in the output directory if it does not exist
+        """
+        cds_root_virtuoso = self.find_cds_root_virtuoso()
+
+        if cds_root_virtuoso is None:
+            contents = """\
+## cds.lib: Defines the locations of compiled libraries.
+softinclude {0}/tools/inca/files/cds.lib
+# needed for referencing the library 'basic' for cells 'cds_alias', 'cds_thru' etc. in analog models:
+# NOTE: 'virtuoso' executable not found!
+# define basic ".../tools/dfII/etc/cdslib/basic"
+define work "{1}/libraries/work"
+""".format(
+                self._cds_root_xrun, self._output_path
+            )
+        else:
+            contents = """\
+## cds.lib: Defines the locations of compiled libraries.
+softinclude {0}/tools/inca/files/cds.lib
+# needed for referencing the library 'basic' for cells 'cds_alias', 'cds_thru' etc. in analog models:
+define basic "{1}/tools/dfII/etc/cdslib/basic"
+define work "{2}/libraries/work"
+""".format(
+                self._cds_root_xrun, cds_root_virtuoso, self._output_path
+            )
+
+        write_file(self._cdslib, contents)
+
+    def setup_library_mapping(self, project):
+        """
+        Compile project using vhdl_standard
+        """
+        mapped_libraries = self._get_mapped_libraries()
+
+        for library in project.get_libraries():
+            self._libraries.append(library)
+            self.create_library(library.name, library.directory, mapped_libraries)
+
+    def compile_source_file_command(self, source_file):
+        """
+        Returns the command to compile a single source file
+        """
+        if source_file.is_vhdl:
+            return self.compile_vhdl_file_command(source_file)
+
+        if source_file.is_any_verilog:
+            return self.compile_verilog_file_command(source_file)
+
+        raise CompileError
+
+    @staticmethod
+    def _vhdl_std_opt(vhdl_standard):
+        """
+        Convert standard to format of xrun command line flag
+        """
+        if vhdl_standard == VHDL.STD_2002:
+            return "-v200x -extv200x"
+
+        if vhdl_standard == VHDL.STD_2008:
+            return "-v200x -extv200x -inc_v200x_pkg"
```

Comment:
I took it from #684 but I'm not sure it's necessary. If yes, must be also applied to VHDL.SDT_2002? Must be applied to Incisive?

Reply:
Ok, I found some answers...

`To successfully compile the code with the VHDL protected types, a second option in addition to -v200x is needed.  Please add -extv200x to your compiler command line.   This switch is not documented.`

`To compile the latest VHDL 2008 constructs, a new command line option, -inc_v200x_pkg, must be used. This is in addition to the -v200x option.`

So these additions seem ok.

Reply:
They are okay for Xcelium, but I'm not so sure about a "true" Incisive installation - after all, the last Incisive release was 2015.
Thinking about it, I wouldn't be surprised if such an old Incisive cannot run a modern vunit at all anymore, so maybe we can and should remove Incisive support completely.
I'll try to check what the status there is.

Reply:
Ok. Taking this into account seems ok to maintain xcellium.py as parent class and incisive.py could heritage from it?

Reply:
> Ok, so Incisive 15.20 doesn't accept `-inc_v200x_pkg` and without it doesn't compile `vunit/vhdl/logging/src/ansi_pkg.vhd` (which I'm sure is just the first file of many with problems).

@cmarqu, did you try running Incisive or did you find it in the documentation?

In case you can run it, would you please try the user_guide for 1993 I proposed in #737? It might be desirable to support running the subset of tests that is compatible with 1993, for those simulators which support VHDL 1993 properly, but don't support VHDL 2008 (yet).

Until now, I believe we had not run into that use case. All other simulators support both 1993 and 2008, or don't support either of them (XSIM cannot handle VUnit's 1993 codebase). However, strictly it should be possible to use VUnit as a "project builder and test runner", despite many utility libraries not being available without 2008 support.